pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -0,0 +1,358 @@
+			reqLogger.Error(err, fmt.Sprintf("Status update for %s failed", currentFAA.Name))
+			return reconcile.Result{}, err
+		}
+
+		return reconcile.Result{}, nil
+	}
+
+	accountID := *gciOut.Account
+	// Add requested aws managed policies to the role
+	awsManagedPolicyNames := []string{}
+	// Add all aws managed policy names to a array
+	for _, policy := range roleMap[currentFAA.Spec.AWSFederatedRoleName].Spec.AWSManagedPolicies {
+		awsManagedPolicyNames = append(awsManagedPolicyNames, policy)
+	}
+	// Get policy arns for managed policies
+	policyArns := createPolicyArns(accountID, awsManagedPolicyNames, true)
+	// Get custom policy arns
+	customPolicy := []string{roleMap[currentFAA.Spec.AWSFederatedRoleName].Spec.AWSCustomPolicy.Name}
+	customerPolArns := createPolicyArns(accountID, customPolicy, false)
+	policyArns = append(policyArns, customerPolArns[0])
+
+	// Attach the requested policy to the newly created role
+	err = r.attachIAMPolices(awsClient, currentFAA.Spec.AWSFederatedRoleName, policyArns)
+	if err != nil {
+		//TODO() role should be deleted here so that we leave nothing behind.
+
+		SetStatuswithCondition(currentFAA, "Failed to attach policies to role", awsv1alpha1.AWSFederatedAccountFailed, awsv1alpha1.AWSFederatedAccountStateFailed)
+		reqLogger.Error(err, fmt.Sprintf("Failed to attach policies to role requested by '%s'", currentFAA.Name))
+		err := r.client.Status().Update(context.TODO(), currentFAA)
+		if err != nil {
+			reqLogger.Error(err, fmt.Sprintf("Status update for %s failed", currentFAA.Name))
+			return reconcile.Result{}, err
+		}
+
+		return reconcile.Result{}, nil
+	}
+	// Mark AWSFederatedAccountAccess CR as Ready.
+	SetStatuswithCondition(currentFAA, "Account Access Ready", awsv1alpha1.AWSFederatedAccountReady, awsv1alpha1.AWSFederatedAccountStateReady)
+	reqLogger.Info(fmt.Sprintf("Successfully applied %s", currentFAA.Name))
+	err = r.client.Status().Update(context.TODO(), currentFAA)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Status update for %s failed", currentFAA.Name))
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// createIAMPolicy creates the IAM policys in AWSFederatedRole inside of our cluster account
+func (r *ReconcileAWSFederatedAccountAccess) createIAMPolicy(awsClient awsclient.Client, afr awsv1alpha1.AWSFederatedRole, afaa awsv1alpha1.AWSFederatedAccountAccess) error {
+	// Same struct from the afr.Spec.AWSCustomPolicy.Statements , but with json tags as captials due to requirements for the policydoc
+	type awsStatement struct {
+		Effect    string                 `json:"Effect"`
+		Action    []string               `json:"Action"`
+		Resource  []string               `json:"Resource,omitempty"`
+		Principal *awsv1alpha1.Principal `json:"Principal,omitempty"`
+	}
+
+	statements := []awsStatement{}
+
+	for _, sm := range afr.Spec.AWSCustomPolicy.Statements {
+		var a awsStatement = awsStatement(sm)
+		statements = append(statements, a)
+	}
```

Comment:
```suggestion
	}	statements := []awsStatement{}

	for _, statement := range role.Spec.AWSCustomPolicy.Statements {
		statements = append(statements, awsStatement(statement))
	}
```

Reply:
idk what happened with the formatting, but this is obviously for lines 246-249, not starting at 249